schematic/utils/general.py

```diff
@@ -69,4 +69,31 @@ def convert_gb_to_bytes(gb: int):
     return: total number of bytes
     """
     bytes_to_return = gb * 1024 * 1024 * 1024
-    return bytes_to_return
\ No newline at end of file
+    return bytes_to_return
+
+
+def entity_type_checking(syn, manifest_id):
+    """
+    check the entity type of the id that needs to be downloaded
+    Args:
+        manifest_id: id of a manifest
+    Return: 
+        type_entity: type of the manifest being returned
+    """
+    # check the type of entity
+    try: 
+        entity_name = syn.get(manifest_id, downloadFile=False)
+    except: 
+        raise(f'Here: {manifest_id}')
+    type_entity = str(type(entity_name))
+
+    if type_entity == "<class 'synapseclient.table.EntityViewSchema'>":
+        return "asset view"
+    elif type_entity == "<class 'synapseclient.entity.Folder'>":
+        return "folder"
+    elif type_entity == "<class 'synapseclient.entity.File'>":
+        return "file"
+    elif type_entity == "<class 'synapseclient.entity.Project'>":
+        return "project"
+    else: 
```

Comment:
So a number of things here.

1. Instead of using `str(type(...))`, you can do something like this: `isinstance(entity_name, synapseclient.table.EntityViewSchema)`. (Read up on `isinstance`)
2. You can return `entity_name.concreteType` if theres no matching entity type.

Reply:
@thomasyu888 Thank you. Using `isinstance` indeed makes my code looks cleaner.